### PR TITLE
Add e2e tests for BindingConditions

### PR DESCRIPTION
**Pull request by Rizwana777**
#### Description:
-E2E tests for binding conditions in snapshotenvironmentbinding_test.go file

#### Link to JIRA Story (if applicable):
https://issues.redhat.com/browse/GITOPSRVCE-730

### CODE REVIEW

**Comment by Rizwana777** [comment]
test failing for -
```
• [TIMEDOUT] [40.438 seconds]
Devsandbox deployment controller tests Testing Devsandbox deployment controller [BeforeEach] should not create an extra DeploymentTarget for a SpaceRequest with an existing DT 
  [BeforeEach] /go/src/github.com/redhat-appstudio/managed-gitops/tests-e2e/appstudio/devsandboxdeployment_test.go:32
  [It] /go/src/github.com/redhat-appstudio/managed-gitops/tests-e2e/appstudio/devsandboxdeployment_test.go:151
```

File: tests-e2e/appstudio/snapshotenvironmentbinding_test.go [hunk 1]
```diff
@@ -1043,6 +1043,141 @@ var _ = Describe("SnapshotEnvironmentBinding Reconciler E2E tests", func() {
...
+				metav1.Condition{
+					Type:    ErrorOccurred,
+					Status:  metav1.ConditionFalse,
+					Reason:  ErrorOccurred,
+					Message: "",
+				}))
+
+		})
+
+		It("test BindingConditions when Components is empty", func() {
```
Comment (jgwest) [inline tests-e2e/appstudio/snapshotenvironmentbinding_test.go:1113]: `when Components is empty`. Is it empty? It seems like we've defined in the SEB

**Review by jgwest — CHANGES_REQUESTED** [review]

File: tests-e2e/appstudio/snapshotenvironmentbinding_test.go (hunk 1 above)
Comment (Rizwana777) [reply tests-e2e/appstudio/snapshotenvironmentbinding_test.go:1113]: This is for`binding.Status.Components` and in SEB we are defining `binding.Spec.Components`

**Comment by Rizwana777** [comment]
@jgwest Updated changes , PTAL

**Review by jgwest — APPROVED** [review]
LGTM, thanks @Rizwana777!

**Pull request merged**